src/serde_utils.rs

```diff
@@ -0,0 +1,92 @@
+use std::fmt;
+use std::marker::PhantomData;
+use serde::de::{Error, Visitor, Deserialize, SeqAccess};
+use serde::ser::{Serializer, SerializeSeq, Serialize};
+
+/// Map to serializeable representation
+pub trait IntoSerializable {
+    type Output;
+    fn into_serializable(self) -> Self::Output;
+}
+
+/// Map from deserialized representation
+pub trait FromDeserialized : Sized {
+    type Input;
+    fn from_deserialized<E>(input: Self::Input) -> Result<Self, E> where E: Error;
+}
+
+
+
+/// Serde combinator. A sequence visitor that maps deserialized elements
+/// lazily; the visitor can also emit new errors if the elements have errors.
+pub struct MappedSequenceVisitor<T, R, F>
+    where F: Fn(T) -> Result<R, &'static str>
+{
+    f: F,
+    marker: PhantomData<fn() -> T>
+}
```

Comment:
How come this doesn't fail with unused type parameter R?

Reply:
It is used in a projection of the assocoated type of the closure trait. Same rule w asc. types just a bit more hidden when it comes to closures.

Reply:
It is used in a projection of the assocoated type of the closure trait. Same rule w asc. types just a bit more hidden when it comes to closures.